inc/admin/profile/personaltokens/namespace.php

```diff
@@ -0,0 +1,183 @@
+		$user_id = $current_user->ID;
+	}
+
+	$user = get_user_by( 'id', $user_id );
+	if ( empty( $user ) ) {
+		wp_die( __( 'Invalid user ID.' ) );
+	}
+	if ( ! current_user_can( 'edit_user', $user_id ) ) {
+		wp_die( __( 'Sorry, you are not allowed to edit this user.' ) );
+	}
+
+	if ( current_user_can( 'edit_users' ) && ! IS_PROFILE_PAGE ) {
+		$submenu_file = 'users.php';
+	} else {
+		$submenu_file = 'profile.php';
+	}
+
+	if ( current_user_can( 'edit_users' ) ) {
+		$parent_file = 'users.php';
+	} else {
+		$parent_file = 'profile.php';
+	}
+}
+
+/**
+ * Render the access token creation page.
+ */
+function render_page() {
+	bootstrap_profile_page();
+
+	$user = get_user_by( 'id', $GLOBALS['user_id'] );
+
+	if ( isset( $_POST['oauth2_action'] ) ) { // WPCS: CSRF OK
+		$error = handle_page_action( $user );
+
+		if ( is_wp_error( $error ) ) {
+			add_action( 'all_admin_notices', function () use ( $error ) {
+				echo '<div class="error"><p>' . esc_html( $error->get_error_message() ) . '</p></div>';
+			} );
+		}
+	}
+
+	$GLOBALS['title'] = __( 'Personal Access Tokens', 'oauth2' );
+	require ABSPATH . 'wp-admin/admin-header.php';
+
+	$tokens = Access_Token::get_for_user( $user );
+	$tokens = array_filter( $tokens, function ( Access_Token $token ) {
+		$client = $token->get_client();
+		return ! empty( $client ) && $client instanceof PersonalClient;
+	});
+	?>
+	<div class="wrap" id="profile-page">
+		<h1><?php esc_html_e( 'Create a Personal Access Token', 'oauth2' ) ?></h1>
+
+		<p><?php esc_html_e( "The WordPress API allows access to your site by external applications. Personal access tokens allow easy access for personal scripts, command line utilities, or during development. Generally, you shouldn't provide these to applications you don't trust, and you should treat them just like passwords.", 'oauth2' ) ?></p>
+
+		<form action="" method="POST">
+			<table class="form-table">
+				<tr>
+					<th scope="row">
+						<label for="token-name">Token name</label>
+					</th>
+					<td>
+						<input
+							class="regular-text"
+							id="token-name"
+							name="name"
+							required="required"
+							type="text"
+						/>
+
+						<p class="description"><?php esc_html_e( 'Give this token a name so you can easily identify it later.', 'oauth2' ) ?></p>
+					</td>
+				</tr>
+			</table>
+
+			<input type="hidden" name="oauth2_action" value="create" />
+
+			<?php wp_nonce_field( 'oauth2_personal_tokens.create' ) ?>
+
+			<p class="buttons">
+				<button class="button-primary"><?php esc_html_e( 'Generate Token', 'oauth2' ) ?></button>
+			</p>
+		</form>
+	</div>
+	<?php
+	require ABSPATH . 'wp-admin/admin-footer.php';
+	exit;
+}
+
+/**
+ * Handle action from a form.
+ */
+function handle_page_action( WP_User $user ) {
+	$action = $_POST['oauth2_action']; // WPCS: CSRF OK
+
+	switch ( $action ) {
+		case 'create':
+			check_admin_referer( 'oauth2_personal_tokens.create' );
+			$data = wp_unslash( $_POST );
```

Comment:
Not sure if I'm into this, there's no hint that this is all unsafe data etc. Can we just sanitize here? looks like we only use `name`